### PR TITLE
Add GAP_pkg_normalizinterface

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

**Comment by fingolfin** [comment]
Weird...
```
[14:00:11] /opt/x86_64-apple-darwin14/x86_64-apple-darwin14/include/c++/7.1.0/x86_64-apple-darwin14/bits/error_constants.h:135:24: error: ‘EOWNERDEAD’ was not declared in this scope
[14:00:11]        owner_dead =     EOWNERDEAD,
[14:00:11]                         ^~~~~~~~~~
[14:00:11] /opt/x86_64-apple-darwin14/x86_64-apple-darwin14/include/c++/7.1.0/x86_64-apple-darwin14/bits/error_constants.h:151:34: error: ‘ENOTRECOVERABLE’ was not declared in this scope
[14:00:11]        state_not_recoverable =    ENOTRECOVERABLE,
[14:00:11]                                   ^~~~~~~~~~~~~~~
```

**Comment by giordano** [comment]
Looks like https://www.mail-archive.com/gcc-bugs@gcc.gnu.org/msg634084.html?  Looks like we're somehow building with gcc instead of llvm?

**Comment by fingolfin** [comment]
Thanks, that was a good hint. Indeed, it was hardcoded to execute binaries with names `gcc` and `g++`. Weird though that this caused no problems for `GAP_pkg_ferret`, which also uses C++ code... I've patched that now to use `clang` and `clang++` on Darwin (I'll worry about the FreeBSD issue later). Now I get
```
[08:07:25] Undefined symbols for architecture x86_64:
[08:07:25]   "___gmpq_clear", referenced from:
```
which is clearly because `-lgmp` is missing and I'll have to analyze that next (esp. why it affects macos but not linux).

File: G/GAP_pkg/bundled/gac-darwin [hunk 1]
```diff
@@ -66,12 +66,12 @@ gap_compiler="${abs_top_builddir}/gap"
...
 # These three should be filled in by the standard autoconf procedures
-c_compiler="gcc -fno-common -DPIC"
-cxx_compiler="g++ -fno-common -DPIC"
+c_compiler="clang -fno-common -DPIC"
+cxx_compiler="clang++ -fno-common -DPIC"
 c_linker="echo static linking not supported ; exit 1 ;"
 
 # These will need special care
-c_dyn_linker="gcc -bundle -lgap"
+c_dyn_linker="clang -bundle -lgap"
```
Comment (giordano) [inline G/GAP_pkg/bundled/gac-darwin:74]: I'd suggest using the more generic `cc` and `c++`, which are good for all platforms

Comment (fingolfin) [reply G/GAP_pkg/bundled/gac-darwin]: Good point, done

**Comment by fingolfin** [comment]
Finally works. Gotta clean up that upstream build system 😂

File: G/GAP_pkg/GAP_pkg_normalizinterface/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+sources = [
+    ArchiveSource("https://github.com/gap-packages/$(name)/releases/download/v$(upstream_version)/$(name)-$(upstream_version).tar.bz2",
+                  "5fd7a1e13bbd48cd119456745f00f8208cd09109c0f6614f0165e1965b92afdd"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd NormalizInterface*
+
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/NormalizInterface.patch
```
Comment (giordano) [inline G/GAP_pkg/GAP_pkg_normalizinterface/build_tarballs.jl:22]: Wait, I'm incredibly confused: the only source is `NormalizInterface` tarball, where is this patch coming from?  And how are the other two files in `G/GAP_pkg/bundled` involved?

**Pull request merged**